core/src/G3Writer.cxx

```diff
@@ -34,6 +34,16 @@ G3Writer::~G3Writer()
 
 void G3Writer::Process(G3FramePtr frame, std::deque<G3FramePtr> &out)
 {
+        // If in Python context, release the GIL while writing out.
+        // Note we must force serialization of the frames, first,
+        // because in some cases serialization requires calling back
+        // out to Python.
+	PyThreadState *_save = nullptr;
+	if (Py_IsInitialized() && PyGILState_Check()) {
+		frame->GenerateBlobs();
```

Comment:
There's no actual harm to calling GenerateBlobs() unconditionally if you want to simplify this conditional (also no benefit really).